modules/core/shared/src/main/scala/scaladex/core/model/Artifact.scala

```diff
@@ -221,28 +221,59 @@ case class Artifact(
     }
 
   // todo: Add tests for this
-  def scastieURL: String = {
-    val tryBaseUrl = "https://scastie.scala-lang.org/try"
-
-    val targetParam = binaryVersion.platform match {
-      case ScalaJs(_) => Some("t" -> "JS")
-      case _          => None
-    }
-
-    val scalaVersionParam = binaryVersion.language match {
-      case Scala(v) => Some("sv" -> v.toString)
-      case _        => None
-    }
-
-    val params: List[(String, String)] = List(
-      "g" -> groupId.value,
-      "a" -> artifactName.value,
-      "v" -> version.toString
-    ) ++ targetParam ++ scalaVersionParam
-
-    params
-      .map { case (k, v) => s"$k=$v" }
-      .mkString(tryBaseUrl + "?", "&", "")
+  // def scastieURL: String = {
+  //   val tryBaseUrl = "https://scastie.scala-lang.org/try"
+
+  //   val targetParam = binaryVersion.platform match {
+  //     case ScalaJs(_) => Some("t" -> "JS")
+  //     case _          => None
+  //   }
+
+  //   val scalaVersionParam = binaryVersion.language match {
+  //     case Scala(v) => Some("sv" -> v.toString)
+  //     case _        => None
+  //   }
+
+  //   val params: List[(String, String)] = List(
+  //     "g" -> groupId.value,
+  //     "a" -> artifactName.value,
+  //     "v" -> version.toString
+  //   ) ++ targetParam ++ scalaVersionParam
+
+  //   params
+  //     .map { case (k, v) => s"$k=$v" }
+  //     .mkString(tryBaseUrl + "?", "&", "")
+  // }
+
+  // def isSbtPlugin: Boolean = platform match {
+  //   case SbtPlugin(_) => true
+  //   case _            => false
+  // }
+
+  def scastieURL: Option[String] = binaryVersion.platform match {
+    case SbtPlugin(_) | ScalaNative(_) => None
+    case _ =>
```

Comment:
Using pattern matching is okay, it is clear and it does what we need.

But I would like to propose an alternative, using a for-comprehension on options:

```scala
val targetParamOpt = binaryVersion.platform match {
  case ScalaJs(_) => Some("t" -> "JS")
  case _          => Some("t" -> "JVM")
}

val scalaVersionParamOpt = binaryVersion.language match {
  case Scala(v) => Some("sv" -> v.toString)
  case _        => None
}

// combine the optinal target parameter and optional scala version parameter to build the URL
for {
  targetParam <- targetParamOpt
  scalaVersionParam <- scalaVersionParamOpt
}  yield {
  val params: List[(String, String)] = List(
    "g" -> groupId.value,
    "a" -> artifactName.value,
    "v" -> version.toString,
    "o" -> projectRef.organization.toString,
    "r" -> projectRef.repository.toString,
    targetParam,
    scalaVersionParam  
  )
  params.map { case (k, v) => s"$k=$v" }.mkString(tryBaseUrl + "?", "&", "")
}
```

Reply:
@adpi2 What do you think about this?

I have used for-comprehension and added the check whether `SbtPlugin` or `ScalaNative` inside targetParam

```
def scastieURL: Option[String] = {
      val tryBaseUrl = "https://scastie.scala-lang.org/try"

      val targetParam = binaryVersion.platform match {
        case ScalaJs(_) => Some("t" -> "JS")
        case SbtPlugin(_) | ScalaNative(_) => return None
        case _ => Some("t" -> "JVM")
      }

      val scalaVersionParam = binaryVersion.language match {
        case Scala(v) => Some("sv" -> v.toString)
        case _        => return None
      }

      for {
        target <- targetParam
        scalaVersion <- scalaVersionParam
      } yield {
        val params: List[(String, String)] = List(
          "g" -> groupId.value,
          "a" -> artifactName.value,
          "v" -> version.toString,
          "o" -> projectRef.organization.toString,
          "r" -> projectRef.repository.toString,
          target,
          scalaVersion
        )
        params.map { case (k, v) => s"$k=$v" }.mkString(tryBaseUrl + "?", "&", "")
      }
  }
```

Reply:
About this:
```scala
val targetParam = binaryVersion.platform match {
    case ScalaJs(_) => Some("t" -> "JS")
    case SbtPlugin(_) | ScalaNative(_) => return None
    case _ => Some("t" -> "JVM")
  }
```
I have two comments:
1. In Scala we try not to use any return. Here you don't need any because the for-comprehension will return None if any of `targetParam` or `scalaVersionParam` is None.
2. The platform param should be "JVM" only if the platform is `JVM`. The default case, which handles `SbtPlugin`, `ScalaNative` and `MillPlugin`, should be `None`.

Reply:
I accidentally used `return`. I apologize for that

Here is updated code for `scastieURL` method-

```
def scastieURL: Option[String] = {
      val tryBaseUrl = "https://scastie.scala-lang.org/try"

      val targetParam = binaryVersion.platform match {
        case ScalaJs(_) => Some("t" -> "JS")
        case JVM => Some("t" -> "JVM")
        case _ => None
      }

      val scalaVersionParam = binaryVersion.language match {
        case Scala(v) => Some("sv" -> v.toString)
        case _        => None
      }

      for {
        target <- targetParam
        scalaVersion <- scalaVersionParam
      } yield {
        val params: List[(String, String)] = List(
          "g" -> groupId.value,
          "a" -> artifactName.value,
          "v" -> version.toString,
          "o" -> projectRef.organization.toString,
          "r" -> projectRef.repository.toString,
          target,
          scalaVersion
        )
        params.map { case (k, v) => s"$k=$v" }.mkString(tryBaseUrl + "?", "&", "")
      }
  }
```